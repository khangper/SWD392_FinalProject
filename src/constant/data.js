export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
