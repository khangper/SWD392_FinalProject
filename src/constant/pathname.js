export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
