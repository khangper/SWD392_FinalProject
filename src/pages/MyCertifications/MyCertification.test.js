import React from "react";
import { renderWithProviders } from "../../constant/test-util";
import MyCertification from "../MyCertifications/MyCertifications";
import { screen } from "@testing-library/react";

describe("MyCertification", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });
  it("renders the MyCertification component", async () => {
    const { container } = renderWithProviders(<MyCertification />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
