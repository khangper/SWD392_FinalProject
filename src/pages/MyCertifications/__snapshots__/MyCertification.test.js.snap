// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCertification renders the MyCertification component 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;
